compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/SingleAbstractMethodLowering.kt

```diff
@@ -137,16 +144,18 @@ class SingleAbstractMethodLowering(val context: CommonBackendContext) : ClassLow
 
             // Construct a class that wraps an invokable object into an implementation of an interface:
             //     class sam$n(private val invokable: F) : Interface { override fun method(...) = invokable(...) }
-            private fun createObjectProxy(superType: IrType, invokableType: IrType): IrClass =
-                implement(superType, listOf(invokableType)) { fields ->
+            private fun createObjectProxy(superType: IrType, invokableType: IrType): IrClass {
+                val invokableClass = invokableType.classifierOrFail.owner as IrClass
+                val function = invokableClass.functions.single { it.name == OperatorNameConventions.INVOKE }
+                return implement(superType, listOf(invokableType), function) { fields ->
```

Comment:
This probably has no effect at all. CallableReferenceLowering would not add any annotations to invoke().

Reply:
Ack. You are right, that doesn't make sense.